src/devices/Charlieplex/CharlieplexSegment.cs

```diff
@@ -0,0 +1,244 @@
+﻿using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Iot.Device.Multiplexer
+{
+    /// <summary>
+    /// Provides support for Charlieplex multiplexing.
+    /// https://wikipedia.org/wiki/Charlieplexing
+    /// </summary>
+    public class CharlieplexSegment : IDisposable
+    {
+        private readonly bool _shouldDispose;
+        private GpioController _controller;
+        private int[] _pins;
+        private int _nodeCount;
+        private Node[] _nodes;
+        private Node _lastNode;
+
+        /// <summary>
+        /// Initializes a new Charlieplex type that can be use for multiplex over a relatively small number of GPIO pins.
+        /// </summary>
+        /// <param name="pins">The set of pins to use.</param>
+        /// <param name="nodeCount">The count of nodes (like LEDs) that will be addressable. If 0, then the Charlieplex maximum is used for the pins provided (n^2-n).</param>
```

Comment:
Seems like most of the time you will want this to be 0 right? is there any scenario where it makes sense to manually specify the nodeCount?

Reply:
nvm I think this was a dumb question I guess.

Reply:
Yes, it will be frequently specified. Once you get past a small number of pins, the numbers get big quickly and won't necessarily match the number of LEDs you are using.